py-polars/polars/lazyframe/frame.py

```diff
@@ -2908,7 +2908,7 @@ def group_by_rolling(
         """
         index_column = parse_as_expression(index_column)
         if offset is None:
-            offset = f"-{_timedelta_to_pl_duration(period)}"
+            offset = _negate_duration(_timedelta_to_pl_duration(period))
```

Comment:
Is this the root of the change? What did we do before and what do we do now?

Reply:
ah no, sorry, this is just a drive-by

the root of the change is in `group_by_dynamic`:
```diff
-            offset = f"-{_timedelta_to_pl_duration(every)}" if period is None else "0ns"
+            offset = _negate_duration(_timedelta_to_pl_duration(every))
```

Reply:
Ah, right. Indeed confusing that we made that parameter dependent on the existence of period.